src/appforms/src/backbone/040-view02field14field_map.js

```diff
@@ -61,50 +61,98 @@ FieldMapView = FieldView.extend({
+     */
+    function handleLocationError(error) {
+      return cb(error);
+    }
+
+    navigator.geolocation.getCurrentPosition(handleLocationSuccess, handleLocationError, { timeout: 20000, enableHighAccuracy: true });
+  },
   onElementShow: function(index) {
     var wrapperObj = this.getWrapper(index);
     var self = this;
 
     var mapCanvas = wrapperObj.find('.fh_map_canvas')[0];
 
-    if($fh.geo){
-      $fh.geo({
-        interval: 0
-      }, function(geoRes) {
-        // Override with geo, otherwise use defaults
-        var location = {
-          lat: geoRes.lat,
-          lon: geoRes.lon
-        };
-        $fh.map({
-          target: mapCanvas,
-          lon: location.lon,
-          lat: location.lat,
-          zoom: self.mapSettings.defaultZoom,
-          draggable: !self.readonly
-        }, function(res) {
-          self.maps[index] = res.map;
+    self.getPosition(function(err, position) {
+      if(err) {
+        $fh.forms.log.e("Error getting location: ", err.message);
+        self.setErrorText(index, err.message);
+      } else {
+        self.clearError(index);
+      }
+      position = position || {};
+      var geoRes = position.coords || {};
 
-          var marker = new google.maps.Marker({
-            position: self.maps[index].getCenter(),
-            map: self.maps[index],
-            draggable: !self.readonly,
-            animation: google.maps.Animation.DROP,
-            title: 'Drag this to set position'
-          });
-          self.markers[index] = marker;
-          self.mapData[index] = {
-            'lat': marker.getPosition().lat(),
-            'long': marker.getPosition().lng(),
-            'zoom': self.mapSettings.defaultZoom
-          };
-          self.onMapInit(index);
-        }, function(err) {
-          $fh.forms.log.e("Error getting map: ", err);
-          self.onMapInit(index);
+      var location = {
+        lat: geoRes.latitude || self.mapSettings.location.lon,
+        lon: geoRes.longitude || self.mapSettings.location.lat
+      };
+
+      /**
+       * Handling a successful map rendering
+       *
+       * @param mapResponse
+       */
+      function handleMapRenderSuccess(mapResponse) {
+        self.maps[index] = mapResponse.map;
+
+        var marker = new google.maps.Marker({
+          position: self.maps[index].getCenter(),
+          map: self.maps[index],
+          draggable: !self.readonly,
+          animation: google.maps.Animation.DROP,
+          title: 'Drag this to set position'
         });
-      });
-    }
+        self.markers[index] = marker;
+        self.mapData[index] = {
+          'lat': marker.getPosition().lat(),
+          'long': marker.getPosition().lng(),
+          'zoom': self.mapSettings.defaultZoom
+        };
+        self.onMapInit(index);
+      }
+
+      /**
+       * Handling Error Rendering Map
+       * @param {string} errorMessage
+       */
+      function handleMapRenderError(errorMessage) {
```

Comment:
I wonder if it might be a little bit cleaner to have this and the function above moved to the parent object instead of here as nested functions. No strong preference here though if you disagree.